chroma_core/models/stratagem.py

```diff
@@ -0,0 +1,174 @@
+    purge_duration_active = models.BooleanField(
+        default=False, help_text="Indicates if the purge should execute at the given interval"
+    )
+
+    states = ["unconfigured", "configured"]
+    initial_state = "unconfigured"
+
+    class Meta:
+        app_label = "chroma_core"
+        ordering = ["id"]
+
+
+class StratagemUnconfiguredAlert(AlertStateBase):
+    default_severity = logging.ERROR
+
+    def alert_message(self):
+        return "Stratagem did not configure correctly"
+
+    class Meta:
+        app_label = "chroma_core"
+
+    def end_event(self):
+        return AlertEvent(
+            message_str="%s started" % self.alert_item,
+            alert_item=self.alert_item.primary_host,
+            alert=self,
+            severity=logging.INFO,
+        )
+
+    def affected_targets(self, affect_target):
+        affect_target(self.alert_item)
+
+
+class ConfigureStratagemTimerStep(Step):
+    def run(self, kwargs):
+        job_log.debug("Create stratagem timer step kwargs: {}".format(kwargs))
+        # Create systemd timer
+
+
+class ConfigureStratagemJob(StateChangeJob):
+    state_transition = StateChangeJob.StateTransition(StratagemConfiguration, "unconfigured", "configured")
+    stateful_object = "stratagem_configuration"
+    stratagem_configuration = models.ForeignKey(StratagemConfiguration)
+
+    display_group = Job.JOB_GROUPS.COMMON
+    display_order = 10
+
+    requires_confirmation = False
+    state_verb = "Configure Stratagem"
+
+    class Meta:
+        app_label = "chroma_core"
+        ordering = ["id"]
+
+    @classmethod
+    def long_description(cls, stateful_object):
+        return help_text["configure_stratagem_long"]
+
+    def description(self):
+        return help_text["configure_stratagem_description"]
+
+    def get_steps(self):
+        steps = [(ConfigureStratagemTimerStep, {})]
+
+        return steps
+
+
+class RunStratagemStep(Step):
+    database = True
+
+    def run(self, kwargs):
+        def _get_body(mount_point):
+            return {
+                "dump_flist": False,
+                "device": {"path": mount_point, "groups": ["size_distribution", "warn_purge_times"]},
+                "groups": [
+                    {
+                        "rules": [
+                            {
+                                "action": "LAT_COUNTER_INC",
+                                "expression": "< size 1048576",
+                                "argument": "smaller_than_1M",
+                            },
+                            {
+                                "action": "LAT_COUNTER_INC",
+                                "expression": "&& >= size 1048576 < size 1048576000",
+                                "argument": "not_smaller_than_1M_and_smaller_than_1G",
+                            },
+                            {
+                                "action": "LAT_COUNTER_INC",
+                                "expression": ">= size 1048576000",
```

Comment:
Should this be bounded between 1GB and 1TB?  
Are there any requirements for striations?

Reply:
I took this from the baked json. Is this not correct?